packages/frontend/editor-ui/src/features/ai/chatHub/components/AgentEditorModal.vue

```diff
@@ -89,16 +89,14 @@ function resetForm() {
 	agentSelectedCredentials.value = {};
 }
 
-// Watch for modal opening
+// Watch for agent ID to change
 watch(
-	() => uiStore.isModalActiveById.agentEditor,
-	(isOpen) => {
-		if (isOpen) {
```

Comment:
`isOpen` changes when credentials modal is opened/closed, and caused the selection to be overridden.